src/installer/corehost/common/pal.windows.cpp

```diff
@@ -106,13 +106,20 @@ void* pal::map_file_readonly(const pal::string_t& path, size_t &length)
 
     void *address = MapViewOfFile(map, FILE_MAP_READ, 0, 0, 0);
 
-    if (map == NULL)
+    if (address == NULL)
     {
         trace::warning(_X("Failed to map file. MapViewOfFile(%s) failed with error %d"), path.c_str(), GetLastError());
         CloseHandle(file);
+        CloseHandle(map);
         return nullptr;
     }
 
+    // The file-handle (file) and mapping object handle (map) can be safely closed
+    // once the file is mapped. The OS keeps the file open if there is an open mapping into the file.
+
+    CloseHandle(file);
+    CloseHandle(map);
+
     return address;
```

Comment:
Nitpick, but this could be written as:

```C
if (address == nullptr)
{
    trace::warning(...);
}

// ...
CloseHandle(file);
CloseHandle(map);

return address;
```

(Slightly shorter code.)